lib/date_time/date_time.c

```diff
@@ -656,65 +116,45 @@ int date_time_uptime_to_unix_time_ms(int64_t *uptime)
 int date_time_now(int64_t *unix_time_ms)
 {
 	int err;
-	int64_t unix_time_ms_prev;
+	struct timespec tp;
 
 	if (unix_time_ms == NULL) {
 		LOG_ERR("The passed in pointer cannot be NULL");
 		return -EINVAL;
 	}
+	if (!date_time_is_valid()) {
+		LOG_WRN("Valid time not currently available");
+		return -ENODATA;
+	}
 
-	unix_time_ms_prev = *unix_time_ms;
-
-	*unix_time_ms = k_uptime_get();
-
-	err = date_time_uptime_to_unix_time_ms(unix_time_ms);
+	err = clock_gettime(CLOCK_REALTIME, &tp);
```

Comment:
I'm thinking if we should move `clock_gettime()` call to `date_time_core.c`. Logically it belongs there. Drawback is that it would be another case where `date_time.c` forwards the call to `date_time_core.c`.

Reply:
Done